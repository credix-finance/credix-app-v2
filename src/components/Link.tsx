import React from "react";
import NextLink from "next/link";
import { Icon, IconName } from "./Icon";

interface LinkProps {
	label?: string;
	icon?: IconName;
	to: string;
	as?: string;
	className?: string;
}

export const Link = ({ label, icon, as, to, className = "", ...props }: LinkProps) => {
	return (
		<NextLink href={to} as={as}>
			<a className={`no-underline font-bold text-inherit text-base ${className}`} {...props}>
				<div className="flex items-center space-x-3">
					<Icon name={icon} className="w-[18px] h-[18px]" />
					<div>{label}</div>
				</div>
			</a>
		</NextLink>
	);
};
