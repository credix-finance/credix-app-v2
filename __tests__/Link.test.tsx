import { Link } from "@components/Link";
import React from "react";
import renderer from "react-test-renderer";

test("Default link", () => {
	const component = renderer.create(<Link to="/" />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test("Link with icon", () => {
	const component = renderer.create(<Link to="/" icon="chevron-left-square" />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
