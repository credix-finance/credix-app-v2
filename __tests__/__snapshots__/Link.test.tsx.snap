// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default link 1`] = `
<a
  className="no-underline font-bold text-action-primary text-base "
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div
    className="flex items-center space-x-3"
  >
    <div
      className="w-full"
    />
  </div>
</a>
`;

exports[`Link with icon 1`] = `
<a
  className="no-underline font-bold text-action-primary text-base "
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div
    className="flex items-center space-x-3"
  >
    <svg
      className="w-[18px] h-[18px] fill-current"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13.707 8.293a1 1 0 0 1 0 1.414L11.414 12l2.293 2.293a1 1 0 0 1-1.414 1.414l-3-3a1 1 0 0 1 0-1.414l3-3a1 1 0 0 1 1.414 0Z"
      />
      <path
        d="M10.357 3h3.286c1.084 0 1.958 0 2.666.058.729.06 1.369.185 1.961.487a5 5 0 0 1 2.185 2.185c.302.592.428 1.233.487 1.961C21 8.4 21 9.273 21 10.357v3.286c0 1.084 0 1.958-.058 2.666-.06.729-.185 1.369-.487 1.961a5 5 0 0 1-2.185 2.185c-.592.302-1.232.428-1.961.487C15.6 21 14.727 21 13.643 21h-3.286c-1.084 0-1.958 0-2.666-.058-.728-.06-1.369-.185-1.96-.487a5 5 0 0 1-2.186-2.185c-.302-.592-.428-1.232-.487-1.961C3 15.6 3 14.727 3 13.643v-3.286c0-1.084 0-1.958.058-2.666.06-.728.185-1.369.487-1.96A5 5 0 0 1 5.73 3.544c.592-.302 1.233-.428 1.961-.487C8.4 3 9.273 3 10.357 3ZM7.854 5.051c-.605.05-.953.142-1.216.276a3 3 0 0 0-1.311 1.311c-.134.263-.226.611-.276 1.216C5.001 8.471 5 9.264 5 10.4v3.2c0 1.137 0 1.929.051 2.546.05.605.142.953.276 1.216a3 3 0 0 0 1.311 1.311c.263.134.611.226 1.216.276.617.05 1.41.051 2.546.051h3.2c1.137 0 1.929 0 2.546-.051.605-.05.953-.142 1.216-.276a3 3 0 0 0 1.311-1.311c.134-.263.226-.611.276-1.216.05-.617.051-1.41.051-2.546v-3.2c0-1.137 0-1.929-.051-2.546-.05-.605-.142-.953-.276-1.216a3 3 0 0 0-1.311-1.311c-.263-.134-.611-.226-1.216-.276C15.529 5.001 14.736 5 13.6 5h-3.2c-1.137 0-1.929 0-2.546.051Z"
      />
    </svg>
    <div
      className="w-full"
    />
  </div>
</a>
`;
